bubblegum/program/src/state/mod.rs

```diff
@@ -50,17 +49,41 @@ impl Voucher {
     }
 }
 
-#[event]
+#[derive(AnchorSerialize, AnchorDeserialize, PartialEq, Eq, Debug, Clone)]
 pub struct NewNFTEvent {
+    pub account_type: AccountType,
     pub version: Version,
     pub metadata: MetadataArgs,
     pub nonce: u64,
 }
 
-#[event]
+#[derive(AnchorSerialize, AnchorDeserialize, PartialEq, Eq, Debug, Clone)]
+pub struct NodeEvent {
+    pub account_type: AccountType,
+    pub version: Version,
+    // TODO: Figure out why we cannot generate JS API if this is `spl_account_compression::Node`.
+    pub node: [u8; 32],
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, PartialEq, Eq, Debug, Clone)]
 pub struct NFTDecompressionEvent {
+    pub account_type: AccountType,
     pub version: Version,
     pub id: Pubkey,
     pub tree_id: Pubkey,
     pub nonce: u64,
 }
+
+#[derive(AnchorSerialize, AnchorDeserialize, PartialEq, Eq, Debug, Clone)]
+pub enum AccountType {
```

Comment:
At first, this `enum` seemed a bit redundant because the structures we're adding the new field to already encode the kind of the even in their identity (i.e. `struct NodeEvent`, `struct NFTDecompressionEvent`, etc). Is the addition meant to effectively generate a discriminant when they are serialized?

If so, WDYT about (like Noah meantioned) renaming this to something like `BubblegumEvent`,and then having it wrap the even data, i.e.:

```rust
pub enum BubblegumEvent {
    ...
    Node(NodeEvent),
    Decompress(NftDecompressionEvent)   
    ...
}
```
As things are right now, there's nothing stopping on particular event to be initialized with a different event type, and all the initializations are explicit. The may be a bit clearer and less error prone.

Reply:
Yeah a constructor would be less error prone.  I don't know if we need a wrapper struct, the `LeafSchemaEvent` struct is already wrapping the `LeafSchema`, seems like adding a lot of wrapping, and I think the constructor solves the issue of manually setting the field.  LMK what you think of my next draft, and yes to confirm its a bit strange but meant to add a one-byte discriminant to the struct.

Reply:
Added constructor in next version.